### DISCUSSION TITLE
How to view the database's structure

### DISCUSSION THREAD

**Discussion by cruvie** [simolus3/drift#2147] [Q&A] [answered]
How to view the database's structure or operate the database with a GUI tool

**Comment by simolus3** [comment, marked as answer]
I know and recommend these two packages:

1. [storage_inspector](https://pub.dev/packages/storage_inspector) can be used to inspect the contents of multiple storage solutions (including shared preferences and drift) in a IntelliJ-based IDE.
2. [drift_db_viewer](https://pub.dev/packages/drift_db_viewer) can be used to inspect the database in your app.

Alternatively, you could use tools like `adb pull` to obtain the database file from the device and then use any sqlite3 inspector you have to view the database.

**Reply by cruvie** [reply]
thanks u so much😇
